pkg/odo/util/cmdutils.go

```diff
@@ -20,7 +21,7 @@ func LogErrorAndExit(err error, context string, a ...interface{}) {
 		if log.IsJSON() {
 
 			// Machine readble error output
-			machineOutput := machineoutput.GenericError{
+			machineOutput := api.GenericError{
```

Comment:
I noticed that error messages for JSON output are written to stderr:

```
❯ annotate-output odo describe component --name cmp-should-not-exist --namespace 5660-describe-component-json -o json
11:32:05 I: Started odo describe component --name cmp-should-not-exist --namespace test -o json
11:32:05 E: {
11:32:05 E:     "message": "no component found with name \"cmp-should-not-exist\" in the namespace \"5660-describe-component-json\""
11:32:05 E: }
11:32:05 I: Finished with exitcode 1
```

I think the JSON error message is the result of the operation and should be written to stdout, so it can be piped and used by other tools. WDYT?

Reply:
Same issue if I purposely run a wrong subcommand. Both the help message and the JSON output are written to stderr:

```
❯ annotate-output odo describe anything -o json
11:37:42 I: Started odo describe anything -o json
11:37:42 E: Error: Subcommand not found, use one of the available commands: component
11:37:42 E: Usage:
11:37:42 E:   odo describe [flags]
11:37:42 E:   odo describe [command]
11:37:42 E: 
11:37:42 E: Available Commands:
11:37:42 E:   component   Describe a component
11:37:42 E: 
11:37:42 E: Flags:
11:37:42 E:   -h, --help   Help for describe
11:37:42 E: 
11:37:42 E: Additional Flags:
11:37:42 E:       --kubeconfig string    Paths to a kubeconfig. Only required if out-of-cluster.
11:37:42 E:   -v, --v Level              Number for the log level verbosity. Level varies from 0 to 9 (default 0).
11:37:42 E:       --vmodule moduleSpec   Comma-separated list of pattern=N settings for file-filtered logging
11:37:42 E: 
11:37:42 E: Use "odo describe [command] --help" for more information about a command.
11:37:42 E: 
11:37:42 E: {
11:37:42 E:     "message": "Subcommand not found, use one of the available commands: component"
11:37:42 E: }
11:37:42 I: Finished with exitcode 1
```

Reply:
> I noticed that error messages for JSON output are written to stderr:
> 
> ```
> ❯ annotate-output odo describe component --name cmp-should-not-exist --namespace 5660-describe-component-json -o json
> 11:32:05 I: Started odo describe component --name cmp-should-not-exist --namespace test -o json
> 11:32:05 E: {
> 11:32:05 E:     "message": "no component found with name \"cmp-should-not-exist\" in the namespace \"5660-describe-component-json\""
> 11:32:05 E: }
> 11:32:05 I: Finished with exitcode 1
> ```
> 
> This is certainly a wider issue related to error handling for machine-readable output, but I think the JSON error message is the result of the operation and should be written to stdout, so it can be piped and used by other tools. WDYT?

We have documented that in case of error, the message is written to stderr:
https://odo.dev/docs/3.0.0/command-reference/json-output

Let's wait for feedback from plugins developers to fix this if necessary

Reply:
> We have documented that in case of error, the message is written to stderr: https://odo.dev/docs/3.0.0/command-reference/json-output
> 
I see. Great to have this documented. I remember I read this already :-)

> Let's wait for feedback from plugins developers to fix this if necessary
> 
Okay, that makes sense.